src/network/networkd-conf.c

```diff
@@ -152,4 +161,11 @@ int config_parse_duid_rawdata(
         *dhcp_duid_len = duid_index;
 
         return 0;
+
+fail:
+        /* Ignore invalid DUID assignment */
+        *duid_type = _DUID_TYPE_INVALID;
+        *dhcp_duid_len = 0;
+        *dhcp_duid_type = 0;
+        return 0;
```

Comment:
Indeed.. Thanks again for the sharp review. I'll work on a fix to hold on to the 'last-known-good' value.